Fix: Join Public Session | Auto Host Kick to become Host

SourceModzZ opened the pull request:
Sorry for the new pull request. I took the "Auto Host Kick" function from the YimMenu extra addon and implemented it in YimMenu. It works great.

creenl1 commented:
I am also in favor of implementing this feature in YimMenu. Hopefully the pull request will be approved

Deadlineem commented:
Upvote, this could be useful in the menu itself.

Tessio commented:
Tested, works perfectly.

This deserves to be added to the menu itself.

Deadlineem commented:
closes #29

Gstxxx reviewed (commented):
Refactored session auto-kick host logic for clarity and efficiency: Simplified conditions, added const to fixed variables, and streamlined player iteration to reduce nesting and improve readability without altering functionality.

SourceModzZ commented:
> you don't need to disable warnings if you remove the comments, anyway, works fine, W commit

Thanks for the tip :-)

SourceModzZ commented:
> Refactored session auto-kick host logic for clarity and efficiency: Simplified conditions, added const to fixed variables, and streamlined player iteration to reduce nesting and improve readability without altering functionality.

Thank you for the revision.

xiaoxiao921 commented on CMakeLists.txt:
```diff
@@ -66,7 +66,7 @@ target_precompile_headers(YimMenu PRIVATE "${SRC_DIR}/common.hpp")
 
 # Warnings as errors
-set_property(TARGET YimMenu PROPERTY COMPILE_WARNING_AS_ERROR ON)
+set_property(TARGET YimMenu PROPERTY COMPILE_WARNING_AS_ERROR OFF)
```
lol

SourceModzZ replied:
I'm not that experienced, so I chose the simplest way so that the error message no longer appears when compiling. Now I know that I just have to remove the comments.

SourceModzZ replied:
I'm not that experienced, so I chose the simplest way so that the error message no longer appears when compiling. Now I know that I just have to remove the comments.

Skorchekd replied:
this work witrhout local online save file??

Skorchekd commented:
will this work without local onlie save filere?
